extensions/ql-vscode/src/pure/time.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Contains an assortment of helper functions for working with time, dates, and durations.
+ */
+
+
+const durationFormatter = new Intl.RelativeTimeFormat('en', {
+  numeric: 'auto',
+});
+
+// All these are approximate, specifically months and years
+const MINUTE_IN_MILLIS = 1000 * 60;
+const HOUR_IN_MILLIS = 60 * MINUTE_IN_MILLIS;
+const DAY_IN_MILLIS = 24 * HOUR_IN_MILLIS;
+const MONTH_IN_MILLIS = 30 * DAY_IN_MILLIS;
+const YEAR_IN_MILLIS = 365 * DAY_IN_MILLIS;
+
+export function humanizeDuration(diffInMs?: number) {
```

Comment:
I think it's worth having comments on these functions to help users of them know what they do.